File: js/platform/src/modules/queue.ts
import { http } from "../core/http.js";
import { Endpoints } from "../core/endpoints.js";
import type { ApiResult } from "../core/result.js";

export const queue = {
  enqueue(body: {
    queueName: string;
    payload?: Record<string, unknown>;
    priority?: number;
    delaySec?: number;
    retries?: number;
    retryDelaySec?: number;
    timeoutSec?: number;
  }): Promise<ApiResult<Record<string, unknown>>> {
    return http.postMap(Endpoints.queueEnqueue, body);
  },

  getStatus(id: string): Promise<ApiResult<Record<string, unknown>>> {
    return http.getMap(Endpoints.queueStatus(id));
  },

  listJobs(query?: { queueName?: string; state?: string }): Promise<ApiResult<unknown[]>> {
    return http.getList(Endpoints.queueJobs, query);
  },
};

File: flutter/platform/lib/src/modules/queue.dart
import '../core/http_client.dart';
import '../core/api_endpoints.dart';
import '../core/result.dart';

/// Queue module — enqueue, status, list jobs.
class QueueModule {
  const QueueModule();

  Future<ApiResult<Map<String, dynamic>>> enqueue(Map<String, dynamic> body) =>
      HttpClient.instance.postMap(ApiEndpoints.queueEnqueue, body: body);

  Future<ApiResult<Map<String, dynamic>>> getStatus(String id) =>
      HttpClient.instance.getMap(ApiEndpoints.queueStatus(id));

  Future<ApiResult<List<dynamic>>> listJobs({
    String? queueName,
    String? state,
  }) =>
      HttpClient.instance.getRawList(
        ApiEndpoints.queueJobs,
        queryParams: {
          if (queueName != null) 'queueName': queueName,
          if (state != null) 'state': state,
        },
      );
}
